rehydrate/shared/expiration/handler.go

```diff
@@ -0,0 +1,113 @@
+package expiration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/pennsieve/rehydration-service/shared/idempotency"
+	"github.com/pennsieve/rehydration-service/shared/s3cleaner"
+	"log/slog"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type Handler struct {
+	idempotencyStore idempotency.Store
+	cleaner          s3cleaner.Cleaner
+	logger           *slog.Logger
+}
+
+func NewHandler(store idempotency.Store, cleaner s3cleaner.Cleaner, logger *slog.Logger) *Handler {
+	return &Handler{
+		idempotencyStore: store,
+		cleaner:          cleaner,
+		logger:           logger,
+	}
+}
+
+func (h *Handler) Handle(ctx context.Context) error {
+	now := time.Now()
+	h.logger.Info("starting expiration check", slog.Time("time", now))
+	toExpire, err := h.idempotencyStore.QueryExpirationIndex(ctx, now, 100)
```

Comment:
100 is the limit on the number of items to return -- seems reasonable to process only that many at a time (and also seems reasonable that there will never be more than 100 expired rehydrations ready to be expired). In the case that there are more than 100 expired rehydrations ready to be expired, those beyond the limit will get picked up the next time the expiration check is run?

Reply:
Thanks @muftring, that's actually bad naming/documentation/design on my part. It should really be called `batchSize` and maybe be a field in the `expiration.Handler` rather than a method argument. The implementation of `QueryExpirationIndex` does check `lastEvaluatedKey` and will issue as many Query calls as necessary to find all expired rehydrations in a single `QueryExpirationIndex` call.